### PR TITLE
Feature/36 projection tool

### PR DESCRIPTION
**Pull request by TimPurdum**
This turned into a large branch, apologies to the reviewers!

1. Migrated the following files from our private extension repo:
    - `DisplayProjection.razor`
    - `LogicComponent.cs`
    - `Projection.cs`
    - `projection.ts`
    - `GeoJSONLayer.cs`
2. Refactored TypeScript files to include `dotNetBuilder.ts` and `jsBuilder.ts` to do type conversions when the basic json serialization fails.
3. Created a new `ProjectionTool.razor` sample page.
4. Created `GeographicTransformation.cs` and `GeographicTransformationSteps.cs` to support the `Projection.GetTransformation(s)` methods.
5. Fixed a bug in `arcGisJsInterop.disposeView` that was breaking double-map layouts.
6. Moved `drawWithGeodesicBufferOnPointer` out of core library and into `..Core.Sample.Shared\wwwroot\js\extensions.js`, and updated `DisplayProjection.razor` to demonstrate how a consumer can create their own javascript extensions that call into the core library.
7. Minor css and json serialization fixes.

### CODE REVIEW

**Comment by TimPurdum** [comment]
image

File: src/dymaptic.GeoBlazor.Core/Objects/GeographicTransformationStep.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace dymaptic.GeoBlazor.Core.Objects;
+
+public class GeographicTransformationStep
+{
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public bool? IsInverse { get; set; }
+    
+    /// <summary>
+    ///     Wkid = Well-Know Id
```
Comment (seahro) [inline src/dymaptic.GeoBlazor.Core/Objects/GeographicTransformationStep.cs:12]: minor issue: Well-Known

**Review by seahro — APPROVED** [review]
Code refactoring and projection adjustments look solid.

**Comment by TimPurdum** [comment]
overload `Projection.Project` to accept a single Geometry or an array.

**Comment by TimPurdum** [comment]
re-add SpatialReference WGS84 and WebMercator as static properties.

**Comment by TimPurdum** [comment]
Closes #36

**Pull request merged**